Sources/HiPMobile/BusinessLayer/Models/ModelClasses/JoinClasses/JoinPagePage.cs

```diff
@@ -0,0 +1,30 @@
+﻿using PaderbornUniversity.SILab.Hip.Mobile.Shared.DataAccessLayer;
+
+namespace PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Models.JoinClasses
+{
+    /// <summary>
+    /// Represents the many-to-many relationship between a
+    /// <see cref="Models.Page"/> and its <see cref="Page.AdditionalInformationPages"/>.
+    /// </summary>
+    public class JoinPagePage : IJoinEntitySameType<Page>
+    {
+        public Page Page { get; set; }
+        public string PageId { get; set; }
+
+        public Page AdditionalInformationPage { get; set; }
+        public string AdditionalInformationPageId { get; set; }
+
+        public Page this[JoinSide side]
```

Comment:
I've never seen this C# feature, could you please give me a hint what it's called so I can understand what this does? Thanks!

Reply:
It's called an indexer. It's most frequently used in collections, so you can e.g. access a list item with `myList[3]` instead of something like `myList.get(3)`.

Reply:
Got it, thanks!